vector-im/element-x-android
# Initial support for member suggestions

## Type of change


## Content

Added code to:

- Listen to RTE updates for mention suggestions.
- Use these updates to filter room members.
- Display the resulting list of possible mentions in a new UI above the RTE.

Also, `BottomSheetScaffold` has [a couple of bugs](https://issuetracker.google.com/issues/306464779) that make it work poorly with nested scrolling components, so I had to fork it and fix it while they are fixed in the upstream component.

Please note the design isn't final, so it might change in the future. That's also why some strings were hardcoded for now, until they're verified and we add them to localazy.

## Motivation and context

Closes #1452.

## Screenshots / GIFs

Included in the PR.

## Tests

- Go to a room (a small one is better).
- Type `@` in the composer.
- The possible mentions will appear, type more characters to make the filtering more precise.
- Remove the mention text or select another part of the message to make the mention suggestion UI disappear.

Note that this only adds the searching for room members and related UI, but it doesn't actually add the selected mention to the RTE.

## Tested devices

- OS version(s): 11

## Checklist
